docs/spec.md

```diff
@@ -199,6 +199,19 @@ quantization parameters. Quantized tensor types have the following constraints:
   * (C12) `quantization_dimension < size(shape)`.
   * (C13) `size(scales) = shape[quantization_dimension]`.
 
+For simplicity, we only define individual operation semantics for per-tensor
+quantized type. Semantics of per-axis type, if supported by the op, can be
+derived as follows:
+
+Given an operation specification with per-tensor quantized type, let the
+semantics of the operation includes some application of the quantization
+parameters `zero_point` and `scale` to a tensor element`e` at index = `[i0, ...,
+id, ..., iR-1]`. The specification for the semantics of the same operation on
+tensor element `e`, with per-axis quantized type with quantization dimension `d`
+, would exactly be the same as the one with per-tensor type except that the
+`zero_point` and `scale` will be replaced with their per-axis counterparts,
+`zero_points[id]` and `scales[id]` resp.
```

Comment:
Could it be easier to always use `zero_points` and `scales`, and if per-tensor quantization is used, then the size of zero_points/scales must be 1, and only zero_points/scales[0] would be used?

Reply:
Hi @eric-k256 
Thanks for pointing that point. I remember that at some point I tried to consider the option and here was my findings:

For per-axis quantization with quantization axis  = d, an element at index `[i0, .., id,..., iR-1]` determines the  values for the `zero_points[id]` and `scales[id]` applied to it. Having said that if we use the per-axis as the default case, while specifying the op semantics, then the semantic need to capture the above relation, which might make the specification hard to follow.  

As an example, for AddOp,
```
* For quantized types:
  * `float_result = (lhs - zero_point(lhs)) * scale(lhs) +
    (rhs - zero_point(rhs)) * scale(rhs)`.
  * `rounded_result = round_nearest_even(float_result / scale(result))`.
  * `result = clamp(storage_min(result), rounded_result + zero_point(result), storage_max(result))`.
```
 
because we are  using per-tensor as default case, we can seamlessly use `scale(lhs)`  and `zero_point(lhs)` without referring to which `lhs` slice it is applied to.  

Please let me know if it makes sense.

Reply:
Yes, it makes sense. It does describe the operation appropriately. Thanks for confirming.